File: android/src/main/java/me/carda/awesome_notifications/notifications/managers/DismissedManager.java
package me.carda.awesome_notifications.notifications.managers;

import android.content.Context;

import com.google.common.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

import me.carda.awesome_notifications.AwesomeNotificationsPlugin;
import me.carda.awesome_notifications.Definitions;
import me.carda.awesome_notifications.notifications.models.returnedData.ActionReceived;
import me.carda.awesome_notifications.notifications.models.returnedData.ActionReceived;

public class DismissedManager {

    private static final SharedManager<ActionReceived> shared
            = new SharedManager<>(
                    "DismissedManager",
                    ActionReceived.class,
                    "ActionReceived");

    public static Boolean removeDismissed(Context context, Integer id) {
        return shared.remove(context, Definitions.SHARED_DISMISSED, id.toString());
    }

    public static List<ActionReceived> listDismissed(Context context) {
        return shared.getAllObjects(context, Definitions.SHARED_DISMISSED);
    }

    public static void saveDismissed(Context context, ActionReceived received) {
        shared.set(context, Definitions.SHARED_DISMISSED, received.id.toString(), received);
    }

    public static ActionReceived getDismissedByKey(Context context, Integer id){
        return shared.get(context, Definitions.SHARED_DISMISSED, id.toString());
    }

    public static void cancelAllDismissed(Context context) {
        List<ActionReceived> receivedList = shared.getAllObjects(context, Definitions.SHARED_DISMISSED);
        if (receivedList != null){
            for (ActionReceived received : receivedList) {
                cancelDismissed(context, received.id);
            }
        }
    }

    public static void cancelDismissed(Context context, Integer id) {
        ActionReceived received = getDismissedByKey(context, id);
        if(received !=null)
            removeDismissed(context, received.id);
    }

    public static void commitChanges(Context context){
        shared.commit(context);
    }
}

File: ios/Classes/lib/core/managers/DismissedManager.swift
//
//  DismissedManager.swift
//  awesome_notifications
//
//  Dismissed by CardaDev on 31/01/22.
//

import Foundation

public class DismissedManager {
    
    static let sharedManager:SharedManager = SharedManager(tag: Definitions.SHARED_DISMISSED)
    
    public static func removeDismissed(id:Int) -> Bool {
        return sharedManager.remove(referenceKey: String(id));
    }

    public static func listDismissed() -> [ActionReceived] {
        var returnedList:[ActionReceived] = []
        let dataList = sharedManager.getAllObjects()
        
        for data in dataList {
            let received:ActionReceived = ActionReceived(nil).fromMap(arguments: data) as! ActionReceived
            returnedList.append(received)
        }
        
        return returnedList
    }

    public static func saveDismissed(received:NotificationReceived) {
        sharedManager.set(received.toMap(), referenceKey: String(received.id!))
    }

    public static func getDismissedByKey(id:Int) -> ActionReceived? {
        guard let data:[String:Any?] = sharedManager.get(referenceKey: String(id)) else {
          return nil
        }
        return ActionReceived(nil).fromMap(arguments: data) as? ActionReceived
    }

    public static func removeAllDismissed() {
        sharedManager.removeAll()
    }

    public static func removeDismissed(id:Int) {
        _ = sharedManager.remove(referenceKey: String(id))
    }
    
}
